homeassistant/components/squeezebox/browse_media.py

```diff
@@ -57,63 +61,125 @@
     "Tracks": MediaType.TRACK,
     "Playlists": MediaType.PLAYLIST,
     "Genres": MediaType.GENRE,
+    "Favorites": MediaType.TRACK,
+}
+
+LIBRARY_ICONS = {
+    "Artists": "html/images/artists.png",
+    "Albums": "html/images/albums.png",
+    "Tracks": "html/images/musicfolder.png",
+    "Playlists": "html/images/playlists.png",
+    "Genres": "html/images/genres.png",
+    "Favorites": "html/images/favorites.png",
 }
 
 BROWSE_LIMIT = 1000
 
 
+def _lms_prefix(player):
+    return (player.generate_image_url_from_track_id("")).split("music/")[0]
+
+
 async def build_item_response(entity, player, payload):
     """Create response payload for search described by payload."""
+
     internal_request = is_internal_request(entity.hass)
 
     search_id = payload["search_id"]
     search_type = payload["search_type"]
 
     media_class = CONTENT_TYPE_MEDIA_CLASS[search_type]
 
-    if search_id and search_id != search_type:
-        browse_id = (SQUEEZEBOX_ID_BY_TYPE[search_type], search_id)
-    else:
-        browse_id = None
+    children = None
 
-    result = await player.async_browse(
-        MEDIA_TYPE_TO_SQUEEZEBOX[search_type],
-        limit=BROWSE_LIMIT,
-        browse_id=browse_id,
-    )
+    if search_type == "Favorites":
+        _command = ["favorites"]
+        _command.extend(
+            [
+                "items",
+                "0",
+                str(BROWSE_LIMIT),
+                f"item_id:{search_id if search_id != "Favorites" else ""}",
+            ]
+        )
 
-    children = None
+        result = await player.async_query(*_command)
```

Comment:
Rather than crafting our own command and calling the API directly with async_query, I will add favorites support to [pysqueezebox](https://github.com/rajlaud/pysqueezebox).